weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/traceServerDirectClient.ts

```diff
@@ -156,6 +158,90 @@ export class DirectTraceServerClient {
     });
   }
 
+  public callsStreamDownload(
```

Comment:
2 comments on this function:
1. It seems like there are two responsibilities: Reading the stream, then downloading the file. This lower level class is intended to just be a thing layer for querying the server. In this case, invoking the file download belongs higher up in the call stack - could be the hooks layer, or even the `traceServerClient`. since it is specific to the application, as opposed to bridging the interface.
2. The implementation seems to be doing something similar to `public callsStreamQuery` - it feels like these should be the same? Maybe you just want to keep it separate to reduce surface area? That is fine, but might be good to comment that we should merge the comment elements since they effectively do the same thing.